### PR TITLE
Better docs for jnp.gradient

### PR DESCRIPTION
**Pull request by jakevdp**
Also remove `skip_params` option from `util.implements`, as this was its last usage.

Rendered preview: https://jax--24025.org.readthedocs.build/en/24025/_autosummary/jax.numpy.gradient.html

Part of #21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1451,14 +1451,67 @@ def ediff1d(ary: ArrayLike, to_end: ArrayLike | None = None,
...
+
+  Args:
+    f: *N*-dimensional array of function values.
+    varargs: optional list of scalars or arrays specifying spacing of
+      function evaluations. Options are:
+
+      - not specified: unit spacing in all dimensions.
+      - a single scalar: constant spacing in all dimensions.
+      - *N* values: specify different spacing in each dimension:
+        - scalar values indicate constant spacing in that dimension.
```
Comment (dfm) [inline jax/_src/numpy/lax_numpy.py:1478]: The docs build seems to fail because of this indentation level here.

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -1451,14 +1451,67 @@ def ediff1d(ary: ArrayLike, to_end: ArrayLike | None = None,
...
     edge_order: int | None = None,
 ) -> Array | list[Array]:
+  """Compute the numerical gradient of a sampled function.
+
+  JAX implementation of :func:`numpy.gradient`.
+
+  The gradient in ``jnp.gradient`` is computed using second-order finite
+  differences across the array of sampled function values. This should not
+  be confused with :func:`jax.grad`, which computes the autodiff-based
+  gradient of a callable function.
```
Comment (dfm) [inline jax/_src/numpy/lax_numpy.py:1468]: Maybe add a link to the intro autodiff tutorial?

**Review by dfm — APPROVED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:1479]: Fixed, thanks!

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py]: Done
